tests/integration/ha_tests/test_smoke.py

```diff
@@ -0,0 +1,146 @@
+from ..helpers import (
+    CHARM_SERIES,
+    get_unit_address,
+    APPLICATION_NAME,
+    get_primary,
+)
+
+from .helpers import (
+    storage_id,
+    is_storage_exists,
+    create_db,
+    check_db
+)
+
+TEST_DATABASE_RELATION_NAME = "test_database"
+DUP_APPLICATION_NAME = "postgres-test-dup"
+
+logger = logging.getLogger(__name__)
+
+
+@pytest.mark.group(1)
+@pytest.mark.abort_on_fail
+async def test_app_removal(ops_test: OpsTest, charm: str):
+    # Deploy the charm.
+    await ops_test.model.deploy(
+        charm,
+        application_name=APPLICATION_NAME,
+        num_units=1,
+        series=CHARM_SERIES,
+        config={"profile": "testing"},
+    )
+
+    # Reducing the update status frequency to speed up the triggering of deferred events.
+    await ops_test.model.set_config({"update-status-hook-interval": "10s"})
+
+    await ops_test.model.wait_for_idle(apps=[APPLICATION_NAME], status="active", timeout=1500)
+    assert ops_test.model.applications[APPLICATION_NAME].units[0].workload_status == "active"
+
+    primary_name = await get_primary(ops_test, ops_test.model.applications[APPLICATION_NAME].units[0].name)
+    host = get_unit_address(ops_test, primary_name)
+    result = requests.get(f"http://{host}:8008/health")
+    assert result.status_code == 200
+
+    storage_id_str = storage_id(ops_test, primary_name)
+
+    # Check if storage exists after application deployed
+    assert await is_storage_exists(ops_test, storage_id_str)
+
+    await ops_test.model.remove_application(APPLICATION_NAME, block_until_done=True, destroy_storage=True)
+
+    # Check if storage removed after application removal
+    assert not await is_storage_exists(ops_test, storage_id_str)
+
+    print(storage_id_str)
+
+
+@pytest.mark.group(1)
+@pytest.mark.abort_on_fail
+async def test_app_force_removal(ops_test: OpsTest, charm: str):
+    # Deploy the charm.
+    await ops_test.model.deploy(
+        charm,
+        application_name=APPLICATION_NAME,
+        num_units=1,
+        series=CHARM_SERIES,
+        config={"profile": "testing"},
+    )
+
+    # Reducing the update status frequency to speed up the triggering of deferred events.
+    await ops_test.model.set_config({"update-status-hook-interval": "10s"})
+
+    await ops_test.model.wait_for_idle(apps=[APPLICATION_NAME], status="active", timeout=1500)
+    assert ops_test.model.applications[APPLICATION_NAME].units[0].workload_status == "active"
+
+    primary_name = await get_primary(ops_test, ops_test.model.applications[APPLICATION_NAME].units[0].name)
+    host = get_unit_address(ops_test, primary_name)
+    result = requests.get(f"http://{host}:8008/health")
+    assert result.status_code == 200
+
+    storage_id_str = storage_id(ops_test, primary_name)
+
+    # Check if storage exists after application deployed
+    assert await is_storage_exists(ops_test, storage_id_str)
+
+    await ops_test.model.remove_application(APPLICATION_NAME, block_until_done=True, force=True, no_wait=True)
```

Comment:
AFAIK, this will remove storage (as it is removed by default on LXD with hostpath storage type (default)), so the next text will be deployed again without any storage. Consider to add destroy_storage=False (but I am not sure if it help).